stdlib/asm/crypto/hashes/keccak_precompile.masm

```diff
@@ -0,0 +1,70 @@
+#! Keccak256 precompile wrapper using event handling for efficient computation.
+#!
+#! This module provides a wrapper function around the keccak precompile event handler,
+#! allowing efficient keccak256 computation by deferring to the native verifier.
+
+use.std::mem
+use.std::crypto::hashes::rpo
+
+# Constants
+# ================================================================================================
+
+
+#! Size of keccak256 hash output in bytes
+const.KECCAK_HASH_SIZE=32
+
+#! Compute keccak256 hash using precompile event handler.
+#!
+#! This function takes a memory slice as input, computes its keccak256 hash using the
+#! precompile event handler, writes the hash to the specified output location, and
+#! returns a cryptographic commitment to the precompile call on the stack.
+#!
+#! The commitment is computed as RPO([RPO(input), RPO(keccak256(input))]) and can be
+#! used for proof verification and precompile call validation.
+#!
+#! Expected stack state:
+#! [ptr_in, len, ptr_out, ...]
+#!
+#! Final stack state:
+#! [commitment, ...]
+#!
+#! Where:
+#! - ptr_in:     memory address where input bytes start
+#! - len:        number of bytes to hash
+#! - ptr_out:    memory address where to write the 32-byte hash
+#! - commitment: RPO hash of [RPO(input), RPO(keccak256(input))]
+export.keccak256_precompile
+    # Emit keccak precompile event - this triggers the event handler which:
+    # 1. Reads input bytes from memory [ptr_in, ptr_in+len)
+    # 2. Computes keccak256 hash
+    # 3. Pushes hash bytes to advice stack
+    # 4. Stores commitment in advice map for prover recovery
+    # => [ptr_in, len, ptr_out, ...]
+    emit.event("miden_stdlib::hash::keccak")
+
+    # Compute RPO hash of the input memory range
+    exec.rpo::hash_memory
```

Comment:
We should probably check that the input address is word-aligned before calling this (I believe it will be checked inside `rpo::hash_memory` as well, but we could return a nicer error from here).

Reply:
Good point, it's at least worth a mention.